python/ray/experimental/channel/shared_memory_channel.py

```diff
@@ -474,12 +475,70 @@ def close(self) -> None:
         Close this channel by setting the error bit on both the writer_ref and the
         reader_ref.
         """
+        print("SANG-TODO close!")
         self._worker.core_worker.experimental_channel_set_error(self._writer_ref)
         if self.is_local_node(self._reader_node_id):
             self.ensure_registered_as_reader()
         self._worker.core_worker.experimental_channel_set_error(self._reader_ref)
 
 
+class _CloseSignal:
+    pass
+
+
+# @DeveloperAPI
+class BufferedRemoteChannel(ChannelInterface):
+
+    def __init__(self, channel: ChannelInterface):
+        self._channel = channel
+        self._background_thread_started = False
+        self.t = None
+        self.queue = None
+
+    def ensure_registered_as_writer(self):
+        """
+        Check whether the process is a valid writer. This method must be idempotent.
+        """
+        return self._channel.ensure_registered_as_writer()
+
+    def ensure_registered_as_reader(self):
+        """
+        Check whether the process is a valid reader. This method must be idempotent.
+        """
+        return self._channel.ensure_registered_as_reader()
+
+    def write(self, value: Any, timeout: Optional[float] = None) -> None:
+        if not self._background_thread_started:
+            self.start()
+            self._background_thread_started = True
+
+        self.queue.put((value, timeout))
```

Comment:
Not sure if buffering at Python would work. I was thinking the buffering probably happens at CPP level.

Reply:
at least self.queue.put seems too slow. exploring other approaches now